verifier/verifier.py

```diff
@@ -228,26 +229,26 @@ def retrieve_models(self, uri, model_dir):
         """
         Function to obtain pre-trained model for verification predictions
         """
-
-        model_file = uri.split('/')[-1]
-        model_file_sl = f'{model_file}_cb_sl'
-        # Create target Directory if don't exist
-        if not os.path.exists(model_dir):
-            os.mkdir(model_dir)
-            logger.info(f'Directory created: {model_dir}')
-            logger.info('Model download started')
-            filename, _ = urllib.request.urlretrieve(uri,
-                                                     filename='{}/{}'.format(model_dir, model_file))
-            logger.info(f'Model {filename} downloaded')
-            try:
-                with tarfile.open(filename) as tar_f:
-                    tar_f.extractall(model_dir)
-
-                return model_dir, model_file, model_file_sl
-            except Exception:
-                return 'Unable to untar model'
-        else:
-            logger.debug(f'Directory {model_dir} already exists, skipping download')
+        with file_locker.FileLocker('model_op.lock'):
```

Comment:
Why do we need this lock if we're only ever using a single verifier instance?

Reply:
The Docker image runs Gunicorn, which spawns multiple (4) worker processes.